src/odemis/driver/xt_client.py

```diff
@@ -90,23 +85,34 @@ def __init__(self, name, role, children, address, daemon=None,
                           "uri is correct and XT server is"
                           " connected to the network. %s" % (address, err))
 
-        # create the scanner child
-        try:
-            if "mb-scanner" in children:
-                kwargs = children["mb-scanner"]
-                if "xttoolkit" in self._swVersion.lower():
-                    self.xt_type = "xttoolkit"
-                else:
-                    raise TypeError("XTtoolkit must be running to instantiate the multi-beam scanner child.")
-            else:
-                kwargs = children["scanner"]
-        except (KeyError, TypeError):
-            raise KeyError("SEM was not given a 'scanner' or 'mb-scanner' child")
+        # Create the scanner type child(ren)
+        # Check if at least one of the required scanner types is instantiated
+        scanner_types = ["scanner", "fib-scanner", "mb-scanner"]  # All allowed scanners types
+        if not any(scanner_type in children for scanner_type in scanner_types):
+            raise KeyError("SEM was not given any scanner type as child."
+                           "On of the types 'scanner', 'fib-scanner' or 'mb-scanner' need to be included as child")
+
+        if "scanner" in children:
+            kwargs = children["scanner"]
+            self._scanner = Scanner(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._scanner)
+
+        if "fib-scanner" in children:
+            kwargs = children["fib-scanner"]
+            self._fib_scanner = FibScanner(parent=self, daemon=daemon, **kwargs)
+            self.children.value.add(self._fib_scanner)
+
         if "mb-scanner" in children:
+            if "scanner" in children:
+                raise NotImplementedError("The combination of both an multi beam scanner and single beam scanner at "
+                                          "the same time is not supported")
+            kwargs = children["mb-scanner"]
+            if "xttoolkit" in self._swVersion.lower():
+                self.xt_type = "xttoolkit"
```

Comment:
@tepals this seems like a part from previous code which is now included in my PR because I needed to reorder the init. Do you think it is ok to delete the .xt_type safely?

Reply:
I think you can remove it, just don't forget to re-run all the test cases if you do.